Sources/XcbeautifyLib/Pattern.swift

```diff
@@ -314,7 +314,7 @@ enum Pattern: String {
     /// $1 = file path
     /// $2 = filename
     /// $3 = reason
-    case compileWarning = #"^(([^:]*):*\d*:*\d*):\swarning:\s(.*)$"#
+    case compileWarning = #"^(?!ld)(([^:]*):*\d*:*\d*):\swarning:\s(.*)$"#
```

Comment:
Just curious, why does this exclude `ld`?

Reply:
Here is an example.

The original regex will also match build log like this (Use Xcode 14.x or enable `-ld64` under Xcode 15) :

```
ld: warning: Could not find or use auto-linked framework 'XYFramework'
Undefined symbols for architecture x86_64:
  "__swift_FORCE_LOAD_$_swiftCompatibility56", referenced from:
      __swift_FORCE_LOAD_$_swiftCompatibility56_$_ABCKit in ABCKit(y.o)
```

Actually, that's not a compile warning and then `formatCompileWarning` will read 2 more addtional lines. So the actual errors will be suppressed.

So I exclude `ld` and print it correctly with `formatLdWarning`.

Reply:
Looks like there is a matching order issue with `ld` related regex expressions.

Once it matches `ldError`, next time `ldWarning` can't be matched as usual. I think I should create a new `ld` related issue and fix it later.

Reply:
Got it. Thanks for the clarification!